src/reactionsystem.jl

```diff
@@ -1548,17 +1605,59 @@ function Base.convert(::Type{<:NonlinearSystem}, rs::ReactionSystem; name = name
                          as_odes = false, include_zero_odes)
-    error_if_constraint_odes(NonlinearSystem, fullrs)
-    eqs, sts, ps, obs, defs = addconstraints!(eqs, fullrs, ists, ispcs; remove_conserved)
+    eqs, us, ps, obs, defs = addconstraints!(eqs, fullrs, ists, ispcs; remove_conserved)
+
+    # Throws a warning if there are differential equations in non-standard format.
+    # Next, sets all differential terms to `0`.
+    nonlinear_convert_differentials_check(rs)
+    eqs = [remove_diffs(eq.lhs) ~ remove_diffs(eq.rhs) for eq in eqs]
 
-    NonlinearSystem(eqs, sts, ps;
+
+    NonlinearSystem(eqs, us, ps;
                     name,
                     observed = obs,
                     defaults = _merge(defaults,defs),
                     checks,
                     kwargs...)
 end
 
+# Finds and differentials in an expression, and sets these to 0.
+function remove_diffs(expr)
+    if Symbolics._occursin(Symbolics.is_derivative, expr)
+        return Symbolics.replace(expr, diff_2_zero)
+    else
+        return expr
+    end
+end
+diff_2_zero(expr) = (Symbolics.is_derivative(expr) ? 0.0 : expr)
+
+# Ideally, when `ReactionSystem`s are converted to `NonlinearSystem`s, any coupled ODEs should be 
+# on the form D(X) ~ ..., where lhs is the time derivative w.r.t. a single variable, and the rhs
+# does not contain any differentials. If this is not teh case, we throw a warning to let the user
+# know that they should be careful.
+function nonlinear_convert_differentials_check(rs::ReactionSystem)
+    for eq in equations(rs)
+        # For each equation (that is not a reaction), checks in order: 
+        # If there is a differential on the right hand side.
+        # If the lhs is not on the form D(...).
+        # If the lhs upper level function is not a differential w.r.t. time.
+        # If the contenct of the differential is not a variable (and nothing more).
+        # If either of this is a case, throws the warning.
+        (eq isa Reaction) && continue
+        if Symbolics._occursin(Symbolics.is_derivative, eq.rhs) ||
+                !Symbolics.istree(eq.lhs) ||
+                !isequal(Symbolics.operation(eq.lhs), Differential(get_iv(rs))) || 
+                (length(arguments(eq.lhs)) != 1) ||
+                !any(isequal(arguments(eq.lhs)[1]), nonspecies(rs))
+            @warn "You are attempting to convert a `ReactionSystem` coupled with differential equations. However, some of these differentials are not of the form `D(x) ~ ...` where: 
+                    (1) The left-hand side is a differential of a single variable with respect to the time independent variable, and
+                    (2) The right-hand side does not contain any differentials.
+                  This is permitted (and all differential will be set to `0`), however, it is recommended to proceed with caution to ensure that the produced nonlinear equation is the desired one."
```

Comment:
I think this should be an error with a keyword to override it if a user really wants to do this. I can't really see, especially for higher order derivatives, that this will generally be correct so we should really make sure users want to do this via forcing them to pass the override keyword.

Reply:
added such a keyword, and made a note of it in the error.